packages/vm/src/environment.rs

```diff
@@ -42,55 +42,84 @@ pub struct GasConfig {
     /// bls12-381 pairing equality check cost
-    pub bls12_381_pairing_equality_cost: u64,
-    /// bls12-381 aggregated pairing equality check cost per point
-    /// (added on top of the base pairing equality check cost)
-    pub bls12_381_aggregated_pairing_equality_cost_per_pair: u64,
+    pub bls12_381_pairing_equality_cost: LinearGasCost,
 }
 
 impl Default for GasConfig {
     fn default() -> Self {
         // Target is 10^12 per second (see GAS.md), i.e. 10^6 gas per µ second.
         const GAS_PER_US: u64 = 1_000_000;
         Self {
-            // ~119 us in crypto benchmarks
-            secp256k1_verify_cost: 119 * GAS_PER_US,
-            // ~233 us in crypto benchmarks
-            secp256k1_recover_pubkey_cost: 233 * GAS_PER_US,
-            // ~374 us in crypto benchmarks
-            secp256r1_verify_cost: 374 * GAS_PER_US,
-            // ~834 us in crypto benchmarks
-            secp256r1_recover_pubkey_cost: 834 * GAS_PER_US,
-            // ~63 us in crypto benchmarks
-            ed25519_verify_cost: 63 * GAS_PER_US,
-            // Gas cost factors, relative to ed25519_verify cost
-            // From https://docs.rs/ed25519-zebra/2.2.0/ed25519_zebra/batch/index.html
-            ed25519_batch_verify_cost: 63 * GAS_PER_US / 2,
-            ed25519_batch_verify_one_pubkey_cost: 63 * GAS_PER_US / 4,
+            // ~96 us in crypto benchmarks
+            secp256k1_verify_cost: 96 * GAS_PER_US,
+            // ~194 us in crypto benchmarks
+            secp256k1_recover_pubkey_cost: 194 * GAS_PER_US,
+            // ~279 us in crypto benchmarks
+            secp256r1_verify_cost: 279 * GAS_PER_US,
+            // ~592 us in crypto benchmarks
+            secp256r1_recover_pubkey_cost: 592 * GAS_PER_US,
+            // ~35 us in crypto benchmarks
+            ed25519_verify_cost: 35 * GAS_PER_US,
+            // Calculated based on the benchmark results for `ed25519_batch_verify_{x}`.
+            ed25519_batch_verify_cost: LinearGasCost {
+                base: 24 * GAS_PER_US,
+                per_item: 21 * GAS_PER_US,
+            },
+            // Calculated based on the benchmark results for `ed25519_batch_verify_one_pubkey_{x}`.
+            ed25519_batch_verify_one_pubkey_cost: LinearGasCost {
+                base: 36 * GAS_PER_US,
+                per_item: 10 * GAS_PER_US,
+            },
             // just assume the production machines have more than 4 cores, so we can half that
-            bls12_381_aggregate_g1_per_point: 16 * GAS_PER_US / 2,
-            bls12_381_aggregate_g2_per_point: 33 * GAS_PER_US / 2,
-            bls12_381_hash_to_g1_cost: 324 * GAS_PER_US,
-            bls12_381_hash_to_g2_cost: 528 * GAS_PER_US,
-            // god i wish i was lying
-            bls12_381_pairing_equality_cost: 1038 * GAS_PER_US,
-            bls12_381_aggregated_pairing_equality_cost_per_pair: 108 * GAS_PER_US,
+            bls12_381_aggregate_g1_cost: LinearGasCost {
```

Comment:
Interestingly, while secp256 and ed25519 got a bit cheaper, bls seems to be significantly slower on the new benchmark server.